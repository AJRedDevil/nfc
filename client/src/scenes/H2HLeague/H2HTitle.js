// npm packages
import React, {Component} from 'react';
import PropTypes from 'prop-types';

// our packages
import LinkTab from '../../components/LinkTab';

class H2HTitle extends Component {
  state = {currentTab: this.props.links[0].path};

  handleTabClick = e => {
    const currentTab = e.target.getAttribute('identifier');
    this.setState({
      ...this.state,
      currentTab,
    });
  };

  render() {
    const {links} = this.props;
    return (
      <div className="tabs">
        <ul>
          {links &&
            links.map(link => (
              <LinkTab
                key={link.text}
                link={link}
                currentTab={this.state.currentTab}
                onTabClick={this.handleTabClick}
              />
            ))}
        </ul>
      </div>
    );
  }
}
H2HTitle.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      text: PropTypes.string,
    })
  ),
};

export default H2HTitle;
