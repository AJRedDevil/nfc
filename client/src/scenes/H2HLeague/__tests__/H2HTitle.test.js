// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

// our packages
import H2HTitle from '../H2HTitle';

test('# Navigation', () => {
  const props = {
    currentTab: 'test',
    setNextTab: () => {},
    links: [{path: '', text: 'home'}, {path: 'test', text: 'TestPage'}],
  };
  const component = renderer.create(
    <MemoryRouter>
      <H2HTitle {...props} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
