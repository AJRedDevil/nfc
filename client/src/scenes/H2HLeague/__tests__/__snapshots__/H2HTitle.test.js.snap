// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Navigation 1`] = `
<div
  className="tabs"
>
  <ul>
    <li
      className=""
    >
      <a
        aria-current="true"
        className=""
        href="/"
        identifier=""
        onClick={[Function]}
        style={Object {}}
      >
        home
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        aria-current={false}
        className={undefined}
        href="/test"
        identifier="test"
        onClick={[Function]}
        style={undefined}
      >
        TestPage
      </a>
    </li>
  </ul>
</div>
`;
